x-pack/plugins/apm/server/lib/metrics/by_agent/java/heap_memory/index.ts

```diff
@@ -52,6 +50,9 @@ const chartBase: ChartBase<HeapMemoryMetrics> = {
 };
 
 export async function getHeapMemoryChart(setup: Setup, serviceName: string) {
-  const result = await fetch(setup, serviceName);
-  return transformDataToMetricsChart<HeapMemoryMetrics>(result, chartBase);
+  const result = (await fetch(setup, serviceName)) as MetricSearchResponse<
```

Comment:
If we don't explicitly tell which keys from `aggregations` are related to the chart, TS will create a union type of date_histogram and min/max buckets, and TS does not allow (AFAICT) maps over union types: https://github.com/Microsoft/TypeScript/pull/29011#issuecomment-451212987